contracts/finance/andromeda-rate-limiting-withdrawals/src/contract.rs

```diff
@@ -33,60 +33,29 @@ pub fn instantiate(
 ) -> Result<Response, ContractError> {
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
 
-    // Can't choose 0 sources for withdrawal frequency
-    require(
-        msg.minimal_withdrawal_frequency.is_some() || msg.contract_key.is_some(),
-        ContractError::UnspecifiedWithdrawalFrequency {},
-    )?;
-
-    // Choose only 1 source for withdrawal frequency
-    if msg.minimal_withdrawal_frequency.is_some() && msg.contract_key.is_some() {
-        return Err(ContractError::OnlyOneSourceAllowed {});
-    };
-
-    if let Some(contract_key) = msg.contract_key {
-        // If key is provided
-        if let Some(key) = contract_key.key {
-            let message = QueryMsg::AndrQuery(AndromedaQuery::Get(Some(to_binary(&key)?)));
-            let resp: GetValueResponse = deps
-                .querier
-                .query_wasm_smart(contract_key.contract_address.clone(), &message)?;
-
-            let minimum_time: Uint128 = resp.value.try_get_uint128()?;
-
-            let coin = CoinAllowance {
+    match msg.minimal_withdrawal_frequency {
+        MinimumFrequency::Time { time } => ALLOWED_COIN.save(
+            deps.storage,
+            &CoinAllowance {
+                coin: msg.allowed_coin.coin,
+                limit: msg.allowed_coin.limit,
+                minimal_withdrawal_frequency: time,
+            },
+        )?,
+        MinimumFrequency::AddressAndKey { address_and_key } => ALLOWED_COIN.save(
+            deps.storage,
+            &CoinAllowance {
                 coin: msg.allowed_coin.clone().coin,
                 limit: msg.allowed_coin.limit,
-                minimal_withdrawal_frequency: minimum_time,
-            };
-
-            ALLOWED_COIN.save(deps.storage, &coin)?;
-        }
-        // If key isn't provided
-        let message = AndromedaQuery::Get(None);
-
-        let resp: GetValueResponse = deps
-            .querier
-            .query_wasm_smart(contract_key.contract_address.clone(), &message)?;
-
-        let minimum_time: Uint128 = resp.value.try_get_uint128()?;
-
-        let coin = CoinAllowance {
-            coin: msg.allowed_coin.clone().coin,
-            limit: msg.allowed_coin.limit,
-            minimal_withdrawal_frequency: minimum_time,
-        };
-
-        ALLOWED_COIN.save(deps.storage, &coin)?;
-    }
-    // If minimum time is directly provided
-    if let Some(minimum_time) = msg.minimal_withdrawal_frequency {
-        let coin = CoinAllowance {
-            coin: msg.allowed_coin.coin,
-            limit: msg.allowed_coin.limit,
-            minimal_withdrawal_frequency: minimum_time,
-        };
-        ALLOWED_COIN.save(deps.storage, &coin)?;
+                minimal_withdrawal_frequency: query_primitive::<GetValueResponse>(
```

Comment:
Yes it's stored on instantiation. Nothing happens since we removed the ability to update the allowed coin. I guess referencing a primitive will prove more useful in other ADOs.